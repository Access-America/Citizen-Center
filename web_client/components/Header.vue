<template>
    <div>
        <!-- Not an official website (yet)
        <div class="px-3 bg-base-lightest">
            <div
                class="flex items-center justify-center py-1 text-xs leading-tight"
            >
                <img
                    src="@hursey013/tailwindcss-uswds/dist/img/us_flag_small.png"
                    alt="Flag"
                />
                <span class="ml-1"
                    >An official website of the United States government</span
                >
            </div>
        </div>
        -->
        <div
            class="flex items-center justify-between mx-3 border-b-1px border-base-lightest"
        >
            <div class="flex items-center space-x-4">
                <img src="~/assets/images/logo.svg" alt="Logo" class="w-3" />
                <h1 class="text-sm font-bold text-gray-90">Citizen Center</h1>
            </div>
            <div>
                <nuxt-link to="Eligibility">Voter Registration</nuxt-link>
            </div>
            <div class="p-2 text-sm text-white uppercase bg-primary">Menu</div>
        </div>
    </div>
</template>

<script>
export default {
    name: 'Header',
}
</script>

<style></style>
