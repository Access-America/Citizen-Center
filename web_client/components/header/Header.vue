<template>
    <div class="bg-white">
        <Overlay :show="mobileNavOpen" />

        <div
            class="flex justify-between pl-2 m-auto sm:pt-4 sm:pr-2 sm:pb-3 navbar max-w-desktop"
        >
            <NuxtLink to="/" class="flex items-center">
                <img
                    class="w-3 h-3 sm:mr-2 sm:w-5 sm:h-5 logo mr-105"
                    src="@/assets/images/logo.svg"
                />
                <span class="font-bold text-ui-sm sm:text-ui-lg md:text-ui-xl">
                    Citizen Center
                </span>
            </NuxtLink>

            <account-buttons class="hidden sm:block" />

            <div
                class="flex items-center block h-6 text-white uppercase cursor-pointer sm:hidden px-105 text-ui-3xs bg-primary"
                @click="openMobileNavMenu"
            >
                Menu
            </div>
        </div>

        <nav-menu :menu="menu" />

        <mobile-nav-menu
            :show="mobileNavOpen"
            :menu="menu"
            :close-menu="closeMobileNavMenu"
        />
    </div>
</template>

<script>
import MobileNavMenu from '@/components/header/MobileNavMenu'
import NavMenu from '@/components/header/NavMenu'
import AccountButtons from '@/components/header/AccountButtons'
import Overlay from '@/components/header/Overlay'
import { menuProps } from '@/components/header/common-proptypes'

export default {
    name: 'Header',
    components: { MobileNavMenu, NavMenu, AccountButtons, Overlay },
    props: {
        menu: menuProps,
    },
    data() {
        return {
            mobileNavOpen: false,
        }
    },
    methods: {
        closeMobileNavMenu() {
            this.mobileNavOpen = false
        },
        openMobileNavMenu() {
            this.mobileNavOpen = true
        },
    },
}
</script>
