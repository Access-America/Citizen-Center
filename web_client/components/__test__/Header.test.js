import { mount } from '@vue/test-utils'
import Header from '~/components/Header.vue'

let wrapper

describe('Header', () => {
    beforeEach(() => {
        // Mount the component
        wrapper = mount(Header)
    })

    it('returns a data object', () => {
        expect(typeof Header.data).toBe('function')
    })
    it('should be a Vue instance', () => {
        // expect(wrapper.isVisible()).toBeTruthy()
        expect(wrapper.find('form').exists()).toBeTruthy()
    })
    it('matches snapshot', () => {
        expect(wrapper.html()).toMatchSnapshot()
    })
})
