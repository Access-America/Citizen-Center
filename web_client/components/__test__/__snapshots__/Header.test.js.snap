// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header matches snapshot 1`] = `
<div><a href="#main-content" class="usa-skipnav">Skip to main content</a>
  <div class="tw-hidden">
    <div class="usa-nav is-visible usa-overlay is-visible"></div>
  </div>
  <div class="usa-overlay"></div>
  <header class="usa-header usa-header--basic">
    <div class="usa-nav-container">
      <div class="usa-navbar">
        <div id="basic-logo" class="usa-logo"><em class="usa-logo__text"><a href="/" title="Home" aria-label="Home"></a></em></div> <button class="usa-menu-btn">
          Menu
        </button>
      </div>
      <nav aria-label="Primary navigation" class="usa-nav"><button class="usa-nav__close"><img src="~/assets/images/close.svg" alt="close"></button>
        <ul class="usa-nav__primary usa-accordion">
          <li class="usa-nav__primary-item"><button aria-expanded="false" aria-controls="basic-nav-section-one" class="usa-accordion__button usa-nav__link"><span>Voting</span></button>
            <ul id="basic-nav-section-one" hidden="hidden" class="usa-nav__submenu">
              <li class="usa-nav__submenu-item"><a href="#">
                  Your National Representatives</a></li>
              <li class="usa-nav__submenu-item"><a href="#"> US Senators</a></li>
              <li class="usa-nav__submenu-item"><a href="#">
                  Members Of US Congress</a></li>
              <li class="usa-nav__submenu-item"><a href="#">
                  State Representatives</a></li>
              <li class="usa-nav__submenu-item"><a href="#">
                  Local Representatives</a></li>
            </ul>
          </li>
          <li class="usa-nav__primary-item"><button aria-expanded="false" aria-controls="basic-nav-section-two" class="usa-accordion__button usa-nav__link"><span>UBI</span></button>
            <ul id="basic-nav-section-two" hidden="hidden" class="usa-nav__submenu">
              <li class="usa-nav__submenu-item"><a href="#">
                  File for Universal Basic Income</a></li>
              <li class="usa-nav__submenu-item"><a href="#">
                  Connect A Bank Account</a></li>
            </ul>
          </li>
        </ul>
        <form role="search" class="usa-search usa-search--small"><label for="basic-search-field-small" class="usa-sr-only">Search small</label> <input id="basic-search-field-small" type="search" name="search" class="usa-input"> <button type="submit" class="usa-button"><span class="usa-sr-only">Search</span></button></form>
      </nav>
    </div>
  </header>
</div>
`;
